File: task.cs
using System;
using System.Linq;

class FizzBuzzDetector
{
    public (string outputString, int count) GetOverlappings(string inputString)
    {
        // Input checking
        if (string.IsNullOrWhiteSpace(inputString) || inputString.Length < 7 || inputString.Length > 100)
        {
            throw new ArgumentException("Input string must be between 7 and 100 characters.");
        }

        // Split words by spaces
        string[] words = inputString.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
        string[] result = new string[words.Length];

        int count = 0;
        int replacementCount = 0;

        for (int i = 0; i < words.Length; i++)
        {
            count++;

            bool isDivisibleBy3 = count % 3 == 0;
            bool isDivisibleBy5 = count % 5 == 0;

            if (isDivisibleBy3 && isDivisibleBy5)
            {
                result[i] = "FizzBuzz";
                replacementCount++;
            }
            else if (isDivisibleBy3)
            {
                result[i] = "Fizz";
                replacementCount++;
            }
            else if (isDivisibleBy5)
            {
                result[i] = "Buzz";
                replacementCount++;
            }
            else
            {
                result[i] = words[i];
            }
        }

        return (string.Join(" ", result), replacementCount);
    }
}

class Program
{
    static void Main()
    {
        FizzBuzzDetector detector = new FizzBuzzDetector();
        string input = "Mary had a little lamb Little lamb, little lamb Mary had a little lamb It's fleece was white as snow";

        var result = detector.GetOverlappings(input);

        Console.WriteLine("Output string:");
        Console.WriteLine(result.outputString);
        Console.WriteLine("\nCount: " + result.count);
    }
}

File: task.js
class FizzBuzzDetector {
    getOverlappings(inputString) {
        // Input checking
        if (!inputString || inputString.length < 7 || inputString.length > 100) {
            throw new Error("Input string must be between 7 and 100 characters");
        }

        // Split words by spaces
        const words = inputString.split(/\s+/);
        const result = [];
        
        let count = 0;
        let replacementCount = 0;
        
        for (let i = 0; i < words.length; i++) {
            
            count++;
            
            const isDivisibleBy3 = count % 3 === 0;
            const isDivisibleBy5 = count % 5 === 0;
            
            if (isDivisibleBy3 && isDivisibleBy5) {
                result.push("FizzBuzz");
                replacementCount++;
            } else if (isDivisibleBy3) {
                result.push("Fizz");
                replacementCount++;
            } else if (isDivisibleBy5) {
                result.push("Buzz");
                replacementCount++;
            } else {
                result.push(words[i]);
            }
        }
        
        return {
            outputString: result.join(' '),
            count: replacementCount
        };
    }
}

// Example of usage
const detector = new FizzBuzzDetector();
const input = `Mary had a little lamb Little lamb, little lamb Mary had a little lamb It's fleece was white as snow`;

const result = detector.getOverlappings(input);
console.log("Output string:");
console.log(result.outputString);
console.log("\nCount:", result.count);


